Rubeus/Commands/Createnetonly.cs

```diff
@@ -12,36 +12,50 @@ public void Execute(Dictionary<string, string> arguments)
         {
             Console.WriteLine("\r\n[*] Action: Create Process (/netonly)\r\n");
 
+            string program = null;
+            string username = null;
+            string password = null;
+            string domain = null;
+            bool show = arguments.ContainsKey["/show"];
+
             if (arguments.ContainsKey("/program"))
             {
-                if (arguments.ContainsKey("/show"))
-                {
-                    if (arguments.ContainsKey("/username"))
-                    {
-                        Helpers.CreateProcessNetOnly(arguments["/program"], true, arguments["/username"], arguments["/domain"], arguments["/password"]);
-                    }
-                    else
-                    {
-                        Helpers.CreateProcessNetOnly(arguments["/program"], true);
-                    }
-                }
-                else
-                {
-                    if (arguments.ContainsKey("/username"))
-                    {
-                        Helpers.CreateProcessNetOnly(arguments["/program"], true, arguments["/username"], arguments["/domain"], arguments["/password"]);
-                    }
-                    else
-                    {
-                        Helpers.CreateProcessNetOnly(arguments["/program"]);
-                    }
-                }
+                program = arguments["/program"];
             }
-
             else
             {
                 Console.WriteLine("\r\n[X] A /program needs to be supplied!\r\n");
+                return;
+            }
+
+            if (arguments.ContainsKey("/username"))
+            {
+                username = arguments["/username"];
+            }
+            if (arguments.ContainsKey("/password"))
+            {
+                password = arguments["/password"];
+            }
+            if (arguments.ContainsKey("/domain"))
+            {
+                domain = arguments["/domain"];
+            }
+
+            if (username == null && password == null && domain == null)
```

Comment:
Could probably do away with this *if* altogether as we're always using the same arguments, just print the domain, username and password being used but I'll let @HarmJ0y comment on that. Other than that, looks good to me.

Reply:
This was to avoid a situation where the user puts in a username and password but no domain and ends up with a randomly generated domain, or some other situation where they forget to put in a certain arg and end up somewhere they didn't expect.

I figured it was desirable to force users to either provide none of the credential values or all of them, rather than have to account for situations where the user might provide some but not others.

Reply:
Personally I'd have done that by doing this:

```
if (arguments.ContainsKey("/username") && arguments.ContainsKey("/password") && arguments.ContainsKey("/domain"))
{
    username = arguments["/username"];
    password = arguments["/password"];
    domain = arguments["/domain"];
}
```

Reply:
Looks good to me, reviewing and landing.